From 48302b7796e7016dc563f3d6366982f073b27ebe Mon Sep 17 00:00:00 2001
From: "Jasper St. Pierre" <jstpierre@mecheye.net>
Date: Sat, 9 Nov 2013 11:43:48 -0500
Subject: [PATCH] goa: Prevent a use-after-free

If the user dismisses the dialog, we'll get an error and free it,
but forget to set it to NULL, and segfault.
---
 gnome-initial-setup/pages/goa/gis-goa-page.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnome-initial-setup/pages/goa/gis-goa-page.c b/gnome-initial-setup/pages/goa/gis-goa-page.c
index 85b8ff5..3228159 100644
--- a/gnome-initial-setup/pages/goa/gis-goa-page.c
+++ b/gnome-initial-setup/pages/goa/gis-goa-page.c
@@ -106,7 +106,7 @@ on_have_providers (GObject       *source,
           gtk_dialog_run (GTK_DIALOG (dialog));
           gtk_widget_destroy (dialog);
         }
-      g_error_free (error);
+      g_clear_error (&error);
     }
 
   g_list_free_full (providers, g_object_unref);
-- 
1.8.3.1

